internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestUtils.java

```diff
@@ -327,11 +339,13 @@ public static void writeBatchData(
       int valueSchemaId,
       CompressionStrategy compressionStrategy,
       Function<String, ByteBuffer> compressionDictionaryGenerator,
-      PubSubProducerAdapterFactory pubSubProducerAdapterFactory) {
+      PubSubProducerAdapterFactory pubSubProducerAdapterFactory,
+      Map<String, String> additionalProperties) {
     Properties props = new Properties();
     props.put(KAFKA_BOOTSTRAP_SERVERS, response.getKafkaBootstrapServers());
     props.setProperty(PARTITIONER_CLASS, response.getPartitionerClass());
     props.putAll(response.getPartitionerParams());
+    props.putAll(additionalProperties);
```

Comment:
nit: for the priority of the additional property you are adding, maybe let's move it one line down to become the last props it is adding to prevent it being overwritten by below configs.

Reply:
Good point, will change it.